### PR TITLE
:sparkles: Demo auth

### LINKED ISSUE #2: Auth
### Describe the feature you'd like to request

- Have a way to login to both the web app and the Expo app
- Set it up in such a way where users can have roles (i.e leads, vps, pit team, drive team) and associated permissions with said role (like MANAGE_ATTENDANCE, MANAGE_SCOUTING_ASSIGNMENTS, etcs)

### Describe the solution you'd like to see

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ThatXliner**
- Closes #2

### CODE REVIEW

File: apps/web/app/page.tsx [hunk 1]
```diff
@@ -1,57 +1,19 @@
...
-    try {
-      createClient(cookieStore)
-      return true
-    } catch (e) {
-      return false
-    }
+export default async function Index() {
+  const cookieStore = cookies();
+  const client = createClient(cookieStore);
+  if ((await client.auth.getUser()).data.user == null) {
```
Comment (realjoshuau) [inline apps/web/app/page.tsx:10]: LGTM :: Could we get a wrapper for this?

**Review by realjoshuau — APPROVED** [review]
LGTM. Was unable to test due to the Supabase instance not sending email, but it _seems_ fine. No mobile auth is fine for now, I'll make another issue regarding that in a bit.

**Comment by ThatXliner** [comment]
> LGTM. Was unable to test due to the Supabase instance not sending email, but it _seems_ fine. No mobile auth is fine for now, I'll make another issue regarding that in a bit.

You don’t need to test on production. Production server should just work if the redirect urls are configured properly. Local Supabase doesn’t send emails

**Comment by ThatXliner** [comment]
> > LGTM. Was unable to test due to the Supabase instance not sending email, but it _seems_ fine. No mobile auth is fine for now, I'll make another issue regarding that in a bit.
> 
> You don’t need to test on production. Production server should just work if the redirect urls are configured properly. Local Supabase doesn’t send emails

On second thought, perhaps only the official hosted Supabase has that email feature built-in to “just work.” Further research is needed

**Comment by realjoshuau** [comment]
> > > LGTM. Was unable to test due to the Supabase instance not sending email, but it _seems_ fine. No mobile auth is fine for now, I'll make another issue regarding that in a bit.
> > 
> > 
> > You don’t need to test on production. Production server should just work if the redirect urls are configured properly. Local Supabase doesn’t send emails
> 
> On second thought, perhaps only the official hosted Supabase has that email feature built-in to “just work.” Further research is needed

Yeah, I have an SMTP server setup on AWS, I just need to hook it up to Supabase, which I'l probably do now or after class today.

**Pull request merged**